ping/rust/src/lib.rs

```diff
@@ -193,3 +201,12 @@ where
 
     Ok(())
 }
+
+pub fn transport_param(client: &testground::client::Client) -> String {
+    client
+        .run_parameters()
+        .test_instance_params
+        .get("transport")
+        .cloned()
+        .unwrap_or_else(|| "tcp".to_owned())
```

Comment:
```suggestion
        .as_deref()
        .unwrap_or("tcp")
```

Just curious whether this would work? (Together with `&str` as a return type).

Reply:
`.get("transport")` returns an `Option<&String>` and `.as_deref()` has no effect on the return type.  From the docs:
```
Converts from Option<T> (or &Option<T>) to Option<&T::Target>.
```

Reply:
If we're looking to avoid the extra allocation, we could `.remove()` (instead of `.get()`) to pull the value out of the hashmap if the value is no longer needed?

Reply:
> `.get("transport")` returns an `Option<&String>` and `.as_deref()` has no effect on the return type. From the docs:
> 
> ```
> Converts from Option<T> (or &Option<T>) to Option<&T::Target>.
> ```

Hmm, the intention was to to convert `&String` to `&str` which _should_ work because `String` implements `Deref<Target = str>`. Seems like we are missing something. It is not important though, I was just curious :)